src/Microsoft.DotNet.ApiCompat/Program.cs

```diff
@@ -17,148 +17,212 @@
         {
-            Settings = StaticSettings;
-            Operands = StaticOperands;
-            AttributeFilter = StaticAttributeFilter;
-        }
+            var app = new CommandLineApplication
+            {
+                Name = "ApiCompat",
+                FullName = "A command line tool to verify that two sets of APIs are compatible.",
+                ResponseFileHandling = ResponseFileHandling.ParseArgsAsSpaceSeparated
+            };
+            app.HelpOption("-?|-h|--help");
+            app.VersionOption("-v|--version", GetAssemblyVersion());
+
+            CommandArgument contracts = app.Argument("contracts", "Comma delimited list of assemblies or directories of assemblies for all the contract assemblies.");
+            contracts.IsRequired();
+            CommandOption implDirs = app.Option("--implDirs", "Comma delimited list of directories to find the implementation assemblies for each contract assembly.", CommandOptionType.SingleValue);
+            implDirs.IsRequired(allowEmptyStrings: true);
+            CommandOption listRules = app.Option("--listRules", "Outputs all the rules. If this options is supplied all other options are ignored.", CommandOptionType.NoValue);
+            CommandOption baseline = app.Option("--baseline", "Baseline file to skip known diffs.", CommandOptionType.SingleValue);
+            CommandOption remapFile = app.Option("--remapFile", "File with a list of type and/or namespace remappings to consider apply to names while diffing.", CommandOptionType.SingleValue);
+            CommandOption skipGroupByAssembly = app.Option("--skipGroupByAssembly", "Skip grouping the differences by assembly instead of flattening the namespaces.", CommandOptionType.NoValue);
+            CommandOption skipUnifyToLibPath = app.Option("--skipUnifyToLibPath", "Skip unifying the assembly references to the loaded assemblies and the assemblies found in the given directories (contractDepends and implDirs).", CommandOptionType.NoValue);
+            CommandOption outFilePath = app.Option("--out", "Output file path. Default is the console.", CommandOptionType.SingleValue);
+            CommandOption resolveFx = app.Option("--resolveFx", "If a contract or implementation dependency cannot be found in the given directories, fallback to try to resolve against the framework directory on the machine.", CommandOptionType.NoValue);
+            CommandOption contractDepends = app.Option("--contractDepends", "Comma delimited list of directories used to resolve the dependencies of the contract assemblies.", CommandOptionType.SingleValue);
+            CommandOption contractCoreAssembly = app.Option("--contractCoreAssembly", "Simple name for the core assembly to use.", CommandOptionType.SingleValue);
+            CommandOption ignoreDesignTimeFacades = app.Option("-idtf|--ignoreDesignTimeFacades", "Ignore design time facades in the contract set while analyzing.", CommandOptionType.NoValue);
```

Comment:
Another common POSIX convention is for short options to only have one letter. If your list of options follows this pattern, then CommandLineUtils will also let you use POSIX clustering. .e.g `git clean -xfd` == `git clean -x -f -d`

```suggestion
            CommandOption ignoreDesignTimeFacades = app.Option("-i|--ignore-design-time-facades", "Ignore design time facades in the contract set while analyzing.", CommandOptionType.NoValue);
```